### PR TITLE
Entity outlines 

### PR DESCRIPTION
**Pull request by SamAins**
Added the Emscripten Bindings to the IndoorrMapEntitiyInfromation

Adding the Outline data to the indoor_map_entity

### CODE REVIEW

File: src/private/emscripten_api/emscripten_indoor_map_entity_information_api.js [hunk 1]
```diff
@@ -133,7 +142,14 @@ function EmscriptenIndoorMapEntityInformationApi(emscriptenApiPointer, cwrap, ru
             var lng = positionLatLngArray[1];
             var position = L.latLng(lat, lng);
 
-            var entity = new indoorMapEntities.IndoorMapEntity(indoorMapEntityId, indoorMapEntityFloorId, position);
+            var Outlines = _GetOutlines(indoorMapEntityModelIds[entityIndex]);
+
+            if(Outlines === null)
```
Comment (jonty-dawson) [inline src/private/emscripten_api/emscripten_indoor_map_entity_information_api.js]: Is this correct? Surely we still want to create an IndoorMapEntity even if it doesn't have outline information available?

File: src/private/emscripten_api/emscripten_indoor_map_entity_information_api.js [hunk 2]
```diff
@@ -144,6 +160,140 @@ function EmscriptenIndoorMapEntityInformationApi(emscriptenApiPointer, cwrap, ru
...
+        );
+
+        var indoorMapEntityNumOfOutlines = _emscriptenMemory.consumeBufferToArray(indoorMapEntityNumOfOutlinesBuf)[0];
+
+        if (!success)
+        {
+            return null;
+        }
+
+        for(var outlineIndex = 0; outlineIndex < indoorMapEntityNumOfOutlines; outlineIndex++)
```
Comment (jonty-dawson) [inline src/private/emscripten_api/emscripten_indoor_map_entity_information_api.js]: I don't understand why we are needing to loop + make multiple calls here. Surely to obtain outline polygons for all entities in this request, we need to
1. call a C method to populate a buffer with information about buffer sizes
2. allocate buffers
3. call a C method to populate those buffers.

File: src/public/indoorMapEntities/indoor_map_entity.js [hunk 3]
```diff
@@ -15,6 +16,13 @@ var IndoorMapEntity = function(indoorMapEntityId, indoorMapFloorId, position) {
         return _position;
     };
 
+    this.getOutlines = function(){
+        return _outlines;
```
Comment (jonty-dawson) [inline src/public/indoorMapEntities/indoor_map_entity.js]: is _outlines guaranteed non-null? If so, is hasOutlines method redundant?

File: src/public/outlines/outline_polygon.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+var OutlinePolygon = function(outerRing, innerRings) {
```
Comment (jonty-dawson) [inline src/public/outlines/outline_polygon.js]: As commented above, suggest not creating new types for OutlinePolygon / OutlinePolygonRing - represent in the same way that we already use for L.Wrld.Polygon instead

File: src/wrld.js [hunk 5]
```diff
@@ -13,8 +13,8 @@ var circle = require("./public/circle.js");
 var heatmap = require("./public/heatmap.js");
 
 require("./private/polyfills.js");
-
-var _baseUrl = "https://cdn-webgl.wrld3d.com/eegeojs/public/latest/";
+// TODO: Remove Before merge
```
Comment (jonty-dawson) [inline src/wrld.js]: TODO

**Comment by SirPryderi** [comment]
Branch already merged -- closing.

**Pull request closed without merging**